0036-spsp-pull-payments/0036-spsp-pull-payments.md

```diff
@@ -0,0 +1,146 @@
+
+### Scope
+
+This document specifies how to conduct a Pull Payment once a STREAM connection has been set up. It is intended for wallet providers and merchants.
+
+### Definitions
+* **Payer** - The entity that units of value are pulled from by the Payee. It is running the [SPSP Server](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Definitions).
+* **Payee** - The entity that is pulling units of value from the Payer. It is running the [SPSP Client](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Definitions).
+* **Pull Payment** - The process of pulling units of value from one account into another account.
+* **Pull Payment Agreement** - An agreement between the Payer and the Payee that specifies how much value can be pulled how often and for how long. 
+* **Pull Payment Pointer** - A [Payment Pointer](../0026-payment-pointers/0026-payment-pointers.md) that includes a unique and opaque string and represents a Pull Payment Agreement. The SPSP Client uses it to query the [SPSP Endpoint](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Definitions) on the SPSP Server. 
+
+### Operation Overview
+
+The Payee's SPSP Client will set up a [STREAM](../0029-stream/0029-stream.md) connection to the Payer's SPSP Server using the Pull Payment Pointer, as described by the [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md). On connection, the SPSP Client tries to pull value from the SPSP server, staying within the terms of the Pull Payment Agreement.
+
+## Model of Operation
+
+### Negotiating the Pull Payment Agreement
+
+Prior to a Pull Payment, the Payer and the Payee have to negotiate a Pull Payment Agreement. This agreement SHOULD be comprised of the parameters defined in the `agreement`-object within the [Response Body](#Response-Body). The Payer's SPSP Server SHOULD store the negotiated Pull Payment Agreement parameters.
+
+The result of the negotiation is a Pull Payment Pointer representing the Pull Payment Agreement. The Pull Payment Pointer is required by the Payee's SPSP Client to connect to the Payer's SPSP server. 
+
+Implementations MAY try to parse the Pull Payment Pointer, however, it MUST support totally opaque Pull Payment Pointers. 
+
+### Conducting the Pull Payment
+
+The Payee's SPSP Client opens a [STREAM](../0029-stream/0029-stream.md) connection to the Payer's SPSP Server as described in the [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md). Note that if there is already an existing open STREAM connection to the `destination_account` the Pull Payment Pointer is resolving to, the connection SHOULD be closed. 
```

Comment:
Should this also say that the server should only allow one stream at a time?

Reply:
You mean specifically saying that? I.e.
> Note that if there is already an existing open STREAM connection to the `destination_account` the Pull Payment Pointer is resolving to, the connection SHOULD be closed. The server SHOULD only allow for one STREAM at a time.

Reply:
:+1: 
`STREAM` -> `stream`